### PR TITLE
update git href regex

### LINKED ISSUE #365: git regex failures
`GitURL` tends to fail on certain edge-case repository URLs.

### PR DESCRIPTION
**Pull request by rokroskar**
# Description

clean up and provide tests for git repo URL regex. 

closes #365

### CODE REVIEW

File: tests/models/test_git.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        },
+        {
+            'href': '../relative/path/to/repo',
+            'pathname': '../relative/path/to/repo'
+        },
+    ]
+)
+def test_valid_href(fields):
+    """Test the various repo regexes."""
+    assert GitURL(**fields) == GitURL.parse(fields['href'])
```
Comment (jirikuncar) [inline tests/models/test_git.py]: ❤️

File: tests/models/test_git.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017, 2018 - Swiss Data Science Center (SDSC)
```
Comment (jirikuncar) [inline tests/models/test_git.py]:
```suggestion
# Copyright 2018 - Swiss Data Science Center (SDSC)
```

File: tests/test_cli.py [hunk 3]
```diff
@@ -641,8 +641,7 @@ def test_show_inputs(tmpdir_factory, project, runner, run):
 
     woop_wc = Path(project) / 'woop.wc'
     assert 0 == run(args=('run', 'wc'), stdin=imported_woop, stdout=woop_wc)
-
-    result = runner.invoke(cli.cli, ['show', 'inputs'])
+    result = runner.invoke(cli.cli, ['show', 'inputs'], catch_exceptions=False)
```
Comment (jirikuncar) [inline tests/test_cli.py]:
```suggestion

    result = runner.invoke(cli.cli, ['show', 'inputs'], catch_exceptions=False)
```

**Comment by leafty** [comment]
Are you following https://github.com/git/git/blob/v2.19.2/Documentation/urls.txt ?

**Comment by rokroskar** [comment]
For future reference, this is the more readable location: https://git-scm.com/docs/git-clone/1.7.12.3

I think we comply except that we don't do anything about `ftp`, but I also don't think that we need to.
